### PR TITLE
Dropped virsh, added libvirt providers

### PR DESCRIPTION
**Pull request by lzap**
Replaced https://github.com/theforeman/theforeman.org/pull/599

### CODE REVIEW

File: _includes/manuals/1.12/3.2.2_installer_options.md [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ For example the hash `{show_diff => true, stringify_facts => false}` for `--pupp
       </tr>
       <tr>
         <td style='white-space:nowrap'>--foreman-proxy-virsh-network</td>
-        <td>Network for virsh DNS/DHCP provider</td>
+        <td>Network for virsh DNS/DHCP provider (deprecated for 1.12+)</td>
```
Comment (domcleal) [inline _includes/manuals/1.12/3.2.2_installer_options.md]: This change isn't in the Puppet module and in fact, the parameter's been renamed. I'd just leave this out until the doc is synced correctly.

File: _includes/manuals/1.12/4.3.11_smartproxy_libvirt.md [hunk 2]
```diff
@@ -1,14 +1,17 @@
 In this chapter, we will describe how to setup DHCP and DNS for use with the
-virsh provider for libvirt.
+libvirt provider for dnsmasq.
 
 This provider is able to change DHCP and DNS settings in libvirt with dnsmasq.
-This is done via the virsh command. These settings can be for only a single
-session, or persistent on the libvirt instance.
+Foreman-proxy process directly connect to libvirt daemon according to
```
Comment (domcleal) [inline _includes/manuals/1.12/4.3.11_smartproxy_libvirt.md]: "The smart proxy directly connects to the libvirt daemon configured"?

File: _includes/manuals/1.12/4.3.11_smartproxy_libvirt.md [hunk 3]
```diff
@@ -62,28 +76,18 @@ Important configuration values are, in `tftp.yml`:
...
 
-    :dhcp: true
-    :dhcp_vendor: virsh
+    :enabled: true
+    :use_provider: dhcp_libvirt
 
-and in `settings.yml`:
+and in `dns_libvirt.yml` and/or `dhcp_libvirt.yml`:
 
     :virsh_network: default
```
Comment (domcleal) [inline _includes/manuals/1.12/4.3.11_smartproxy_libvirt.md]: `:network`

File: _includes/manuals/1.12/4.3.4_smartproxy_dhcp.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ Builtin providers are:
 
 * `dhcp_isc` - ISC DHCP server over OMAPI
 * `dhcp_ms_native` - Microsoft Active Directory using netsh
-* `dhcp_virsh` - libvirt embedded DHCP (dnsmasq)
+* `dhcp_libvirt` - dnsmasq DHCP via libvirt API
```
Comment (domcleal) [inline _includes/manuals/1.12/4.3.4_smartproxy_dhcp.md:9]: Swap lines so it's alphabetical?

File: _includes/manuals/1.12/4.3.4_smartproxy_dhcp.md [hunk 5]
```diff
@@ -66,10 +66,22 @@ The ms_native provider manages reservations in Microsoft Active Directory by run
...
+Possible configuration options in `dhcp_virsh.yml` are:
+
+    # Libvirt network. Only one network is supported.
+    :network: default
+
+    # Libvirt connection. Make sure proxy effective user have permission to connect.
+    :url: qemu:///system
+
+When configuring local or remote connections, make sure the `foreman-proxy`
+effective user have UNIX permissions to libvirt socket or ssh keys are
```
Comment (domcleal) [inline _includes/manuals/1.12/4.3.4_smartproxy_dhcp.md]: s/have/has/

File: _includes/manuals/1.12/4.3.5_smartproxy_dns.md [hunk 6]
```diff
@@ -73,10 +73,22 @@ The **dns_server** option is used if the Smart Proxy is not located on the same
...
+Possible configuration options in `dns_virsh.yml` are:
+
+    # Libvirt network. Only one network is supported.
+    :network: default
+
+    # Libvirt connection. Make sure proxy effective user have permission to connect.
+    :url: qemu:///system
+
+When configuring local or remote connections, make sure the `foreman-proxy`
+effective user have UNIX permissions to libvirt socket or ssh keys are
```
Comment (domcleal) [inline _includes/manuals/1.12/4.3.5_smartproxy_dns.md]: ditto

File: _includes/manuals/1.12/3.2.2_installer_options.md (hunk 1 above)
Comment (lzap) [inline _includes/manuals/1.12/3.2.2_installer_options.md]: You mean to leave the change out of the patch, do you? Removed.

File: _includes/manuals/1.12/4.3.11_smartproxy_libvirt.md (hunk 3 above)
Comment (lzap) [inline _includes/manuals/1.12/4.3.11_smartproxy_libvirt.md]: Good catch

**Comment by lzap** [comment]
Done editing, thanks.

**Comment by mmoll** [comment]
merged, díky @lzap!

**Pull request merged**